internal/config/config_test.go

```diff
@@ -36,6 +37,78 @@ func TestDefaultConfig(t *testing.T) {
 	assert.NotNil(def)
 }
 
+func TestSettingLatestAsVersion(t *testing.T) {
+	testCases := map[string]struct {
+		config     map[string]interface{}
+		configName string
+		wantResult *Config
+		wantErr    bool
+	}{
+		"mix of latest and uint as version value": {
+			config: func() map[string]interface{} {
+				conf := Default()
+				// modify versions as string
+				m := getConfigAsMap(conf, t)
+				m["attestation"].(map[string]interface{})["azureSEVSNP"].(map[string]interface{})["microcodeVersion"] = "latest"
+				m["attestation"].(map[string]interface{})["azureSEVSNP"].(map[string]interface{})["teeVersion"] = "latest"
+				m["attestation"].(map[string]interface{})["azureSEVSNP"].(map[string]interface{})["snpVersion"] = "latest"
+				m["attestation"].(map[string]interface{})["azureSEVSNP"].(map[string]interface{})["bootloaderVersion"] = 1
+				return m
+			}(),
+
+			configName: constants.ConfigFilename,
+			wantResult: func() *Config {
+				conf := Default()
+				conf.Attestation.AzureSEVSNP.BootloaderVersion = 1
+				return conf
+			}(),
+		},
+		"refuse invalid version value": {
+			config: func() map[string]interface{} {
+				conf := Default()
+				m := getConfigAsMap(conf, t)
+				m["attestation"].(map[string]interface{})["azureSEVSNP"].(map[string]interface{})["microcodeVersion"] = "1a"
+				return m
+			}(),
+			configName: constants.ConfigFilename,
+			wantErr:    true,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			require := require.New(t)
+
+			fileHandler := file.NewHandler(afero.NewMemMapFs())
+			if tc.config != nil {
+				require.NoError(fileHandler.WriteYAML(tc.configName, tc.config, file.OptNone))
+			}
+			result, err := fromFile(fileHandler, tc.configName)
+
+			if tc.wantErr {
+				assert.Error(err)
+			} else {
+				require.NoError(err)
+				assert.Equal(tc.wantResult, result)
+			}
+		})
+	}
+}
+
+// getConfigAsMap returns a map of the attestation section of the config.
+func getConfigAsMap(conf *Config, t *testing.T) map[string]interface{} {
+	d, err := yaml.Marshal(&conf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var m map[string]interface{}
+	err = yaml.Unmarshal(d, &m)
+	if err != nil {
```

Comment:
```suggestion
	if if err := yaml.Unmarshal(d, &m); err != nil {
```
also, a bit more descriptive variable names would improve readability here IMO